lucene/core/src/java/org/apache/lucene/store/ByteBufferIndexInput.java

```diff
@@ -107,6 +122,24 @@ public final void readBytes(byte[] b, int offset, int len) throws IOException {
     }
   }
 
+  @Override
+  public void readLongs(ByteOrder byteOrder, long[] dst, int offset, int length) throws IOException {
+    try {
+      final int position = curBuf.position();
+      guard.getLongs(curLongBufferViews[position & 0x07].position(position >>> 3), dst, offset, length);
```

Comment:
For the record, there are 8, not 63 clones. Currently blocks all take `bitsPerValue * 16 + 1` bytes, so adding padding would waste 7 bytes all the time. This would be a 42% increase for 1 bpv blocks and 21% for 2 bpv, which are not that uncommon bits per value for term freqs, especially when exceptions help lower the number of bits per value.

Reply:
OK I must be a little confused here -- the loop above under the lazy init `if` (`for (int i = 0; i < Math.min(Long.BYTES, curBuf.limit()); ++i) {`) seems like it is initializing up to `64` clones, yet the array access here (`position & 0x07`) seems to only use the first `8`.

And I thought the padding would only be needed at the start of postings lists, and only for terms whose frequency is `>= 128`, not at the start of each block, but you're right, since we have the one extra one leading byte (to tell us `numBits`) for each block, pad bytes would indeed need to be for each block, hrmph.  Nevermind ;)

I'm also surprised this trick actually helps -- under the hood the CPU must still do unaligned long decodes (which I think modern X86-64 are good at)?  Maybe add a comment about why this trick is worthwhile?

Reply:
OK, duh!  `Long.BYTES` is 8 :)